### ISSUE TITLE
Old L1T tags consumed in DQM modules

### ISSUE THREAD

**Issue by tvami** [cms-sw/cmssw#37016]
This is the main CMSSW version of 
https://github.com/cms-l1t-offline/cmssw/issues/958
now focusing on the DQM packages.

As of now the following packages were identified to use (very possibly) deprecated L1T records. The list might be updated in the future.

Possible Group | Module | Record | Removed Tag
-- | -- | -- | --
DQM/JetMET | METAnalyzer | L1GtTriggerMenuRcd | L1GtTriggerMenu_CRAFT09_hlt
DQM/JetMET | JetAnalyzer | L1GtTriggerMenuRcd | L1GtTriggerMenu_CRAFT09_hlt
DQM/BPH | BPHMonitor | L1GtStableParametersRcd | L1GtStableParameters_CRAFT09_hlt
DQM/Strips | SiStripMonitorCluster | L1GtStableParametersRcd | L1GtStableParameters_CRAFT09_hlt
DQM/Muons | MuonMonitor | L1GtStableParametersRcd | L1GtStableParameters_CRAFT09_hlt

**Comment by tvami** [comment]
assign dqm

**Comment by tvami** [comment]
This PR https://github.com/cms-sw/cmssw/pull/36986 might be helpful in how to handle the situation (that was for a different record)

**Comment by tvami** [comment]
Hi @arossi83 @sroychow could you please have a look at why `SiStripMonitorCluster` is accessing the `L1GtStableParametersRcd`?

**Comment by mmusich** [comment]
@tvami 
`SiStripMonitorCluster` is NOT accessing directly `L1GtStableParametersRcd` as a very quick search in DXR will demonstrate to you: https://cmssdt.cern.ch/dxr/CMSSW/search?q=L1GtStableParametersRcd&case=true.
On the other hand, `L1GtTriggerMenuRcd` does (since it's a dependent record on `L1GtStableParametersRcd`):

https://github.com/cms-sw/cmssw/blob/95eed0c02185c99501d2355a751357b0bb5db934/CondFormats/DataRecord/interface/L1GtTriggerMenuRcd.h#L32-L36

and that in turn is consumed in `GenericTriggerEventFlag`

https://github.com/cms-sw/cmssw/blob/81ade48beaa46b6c62f174bb678ba44089d73c02/CommonTools/TriggerUtils/src/GenericTriggerEventFlag.cc#L135-L137

and this object is finally constructed in `SiStripMonitorCluster` in:

https://github.com/cms-sw/cmssw/blob/95eed0c02185c99501d2355a751357b0bb5db934/DQM/SiStripMonitorCluster/src/SiStripMonitorCluster.cc#L63-L68

In summary, if you want to loose the dependency on that record, you should not ask the DQM people, but rather either the HLT (as responsibles of `CommonTools/TriggerUtils` or the L1T for the record `L1GtTriggerMenuRcd`).
Finally please notice that in the `GenericTriggerEventFlag`  the consumes is encapsulated in a statement regarding it begin stage1 (pre 2016), so the dependency seems legit to me.

**Comment by tvami** [comment]
Hi @mmusich thanks a lot for looking into this. I see that https://github.com/cms-sw/cmssw/blob/master/DQM/SiStripMonitorClient/python/pset4GenericTriggerEventFlag_cfi.py doesnt have the stage2 boolean on true (and the default is false), so I guess that's the cause of the problem.

**Comment by tvami** [comment]
assign hlt

**Comment by tvami** [comment]
unassign dqm

**Comment by mmusich** [comment]
> doesnt have the stage2 boolean on true (and the default is false), so I guess that's the cause of the problem.

right, probably one just needs to change here (and everywhere else that objected is used) the default for stage2 using the process modifier.

**Comment by tvami** [comment]
Hi @missirol can you make this modification that if the stage-2 era is used then 
https://github.com/cms-sw/cmssw/blob/81ade48beaa46b6c62f174bb678ba44089d73c02/CommonTools/TriggerUtils/src/GenericTriggerEventFlag.cc#L102
is set to true? I think it's better to do this at the source here instead of each single module that's using this flag, do you agree?

**Comment by missirol** [comment]
Based on https://github.com/cms-sw/cmssw/pull/28592/files#r358835228 , my understanding is that the default value of `stage2` should be `false` as it already is in `GenericTriggerEventFlag`, so I don't see any change to be applied to the latter (other than eventually use `fillDescriptions`, but that's purely technical); the cpp class has to handle both cases, is configurable, and defaults to `stage2=false`.

> right, probably one just needs to change here (and everywhere else that objected is used) the default for stage2 using the process modifier.

I would follow this approach, and if i understand Marco's comment, this means changing the python configuration of all the relevant DQM modules with the relevant modifier (that code is mostly under DQM, I think).

**Comment by tvami** [comment]
I was not suggesting to modify the cc code, I was suggesting to have a cff file that changes `stage2` to true if the stage2L1Trigger era is used, something like 

```
from Configuration.Eras.Modifier_stage2L1Trigger_cff import stage2L1Trigger
stage2L1Trigger.toModify(GenericTriggerEventFlag, stage2 = true)
```

Do you think that's a possible solution?

**Comment by missirol** [comment]
Yes, sure. The problem is that I don't see a central `cfi` for the PSet of `GenericTriggerEventFlag`. One way would be to have that `cfi`, it would contain the proc-modifier call as you wrote, and all the client modules would clone from that. Instead, it looks like these modules are creating this PSet from "scratch". So to me, this still makes it a client problem (e.g. "assign dqm"). If you'd like me to look more into the details, it would be helpful to suggest a timeline ("by pre6?").

**Comment by mmusich** [comment]
@tvami 
as far as I can tell one of the `GenericTriggerEventFlag` is already been taken care of here:

https://github.com/cms-sw/cmssw/blob/69c0b8342053669d51c832c0f5053a8e796a5a50/DQM/SiStripMonitorClient/python/SiStripSourceConfigTier0_cff.py#L71-L79

I guess one just needs to adjust the other ones...
How can one check if the dependency is lost after the changes?

**Comment by tvami** [comment]
Here is what we've been using 
```
runTheMatrix.py --what standard -l 138.4 --command "--conditions 121X_dataRun3_L1TriggerTagRemoval_v1 -n 1000" -t 8
```
Sorry it's in 121X, the whole cleaning started back then...

**Comment by mmusich** [comment]
OK.
I think this should solve:
```diff
diff --git a/DQM/SiStripMonitorClient/python/SiStripSourceConfigTier0_cff.py b/DQM/SiStripMonitorClient/python/SiStripSourceConfigTier0_cff.py
index 2e7fd85237c..6cfd463b334 100644
--- a/DQM/SiStripMonitorClient/python/SiStripSourceConfigTier0_cff.py
+++ b/DQM/SiStripMonitorClient/python/SiStripSourceConfigTier0_cff.py
@@ -75,7 +75,9 @@ stage2L1Trigger.toModify(SiStripMonitorClusterBPTX,
         l1tAlgBlkInputTag = cms.InputTag("gtStage2Digis"),
         l1tExtBlkInputTag = cms.InputTag("gtStage2Digis"),
         ReadPrescalesFromFile = cms.bool(True)
-    )
+    ),
+    PixelDCSfilter = dict(stage2 = cms.bool(True)),
+    StripDCSfilter = dict(stage2 = cms.bool(True))
 )
 
 # refitter ### (FIXME rename, move)
diff --git a/DQMOffline/Trigger/python/BPHMonitor_cfi.py b/DQMOffline/Trigger/python/BPHMonitor_cfi.py
index 71dab23a1a8..5f1bab34a06 100644
--- a/DQMOffline/Trigger/python/BPHMonitor_cfi.py
+++ b/DQMOffline/Trigger/python/BPHMonitor_cfi.py
@@ -4,9 +4,6 @@ from DQMOffline.Trigger.bphMonitoring_cfi import bphMonitoring as _bphMonitoring
 
 hltBPHmonitoring = _bphMonitoring.clone()
 
-from Configuration.Eras.Modifier_stage2L1Trigger_cff import stage2L1Trigger
-stage2L1Trigger.toModify(hltBPHmonitoring, stageL1Trigger = 2)
-
 #hltBPHmonitoring.options = cms.untracked.PSet(
 #    SkipEvent = cms.untracked.vstring('ProductNotFound')
 #)
@@ -117,3 +114,9 @@ hltBPHmonitoring.denGenericTriggerEventPSet.hltPaths  = cms.vstring( "HLT_Mu7p5_
 hltBPHmonitoring.denGenericTriggerEventPSet.andOrDcs      = cms.bool( False )
 hltBPHmonitoring.denGenericTriggerEventPSet.errorReplyDcs = cms.bool( True )
 hltBPHmonitoring.denGenericTriggerEventPSet.verbosityLevel = cms.uint32(0)
+
+from Configuration.Eras.Modifier_stage2L1Trigger_cff import stage2L1Trigger
+stage2L1Trigger.toModify(hltBPHmonitoring,
+                         stageL1Trigger = 2,
+                         numGenericTriggerEventPSet = dict(stage2 = cms.bool(True)),
+                         denGenericTriggerEventPSet = dict(stage2 = cms.bool(True)))
diff --git a/DQMOffline/Trigger/python/JetMonitor_cfi.py b/DQMOffline/Trigger/python/JetMonitor_cfi.py
index 932838c5344..b0c5447e221 100644
--- a/DQMOffline/Trigger/python/JetMonitor_cfi.py
+++ b/DQMOffline/Trigger/python/JetMonitor_cfi.py
@@ -39,3 +39,7 @@ hltJetMETmonitoring.denGenericTriggerEventPSet.andOrDcs      = cms.bool( False )
 hltJetMETmonitoring.denGenericTriggerEventPSet.errorReplyDcs = cms.bool( True )
 hltJetMETmonitoring.denGenericTriggerEventPSet.verbosityLevel = cms.uint32(1)
 
+from Configuration.Eras.Modifier_stage2L1Trigger_cff import stage2L1Trigger
+stage2L1Trigger.toModify(hltJetMETmonitoring,
+                         numGenericTriggerEventPSet = dict(stage2 = cms.bool(True)),
+                         denGenericTriggerEventPSet = dict(stage2 = cms.bool(True)))
diff --git a/DQMOffline/Trigger/python/METMonitor_cfi.py b/DQMOffline/Trigger/python/METMonitor_cfi.py
index 8f72ba6acc0..be521b4aad8 100644
--- a/DQMOffline/Trigger/python/METMonitor_cfi.py
+++ b/DQMOffline/Trigger/python/METMonitor_cfi.py
@@ -37,3 +37,8 @@ hltMETmonitoring.denGenericTriggerEventPSet.errorReplyDcs = cms.bool( True )
 hltMETmonitoring.denGenericTriggerEventPSet.verbosityLevel = cms.uint32(1)
 #hltMETmonitoring.denGenericTriggerEventPSet.hltPaths      = cms.vstring("HLT_IsoMu27_v*");
 hltMETmonitoring.denGenericTriggerEventPSet.hltPaths      = cms.vstring();
+
+from Configuration.Eras.Modifier_stage2L1Trigger_cff import stage2L1Trigger
+stage2L1Trigger.toModify(hltMETmonitoring,
+                         numGenericTriggerEventPSet = dict(stage2 = cms.bool(True)),
+                         denGenericTriggerEventPSet = dict(stage2 = cms.bool(True)))
diff --git a/DQMOffline/Trigger/python/MuonMonitor_cfi.py b/DQMOffline/Trigger/python/MuonMonitor_cfi.py
index 93c16eb01c4..cfc74964fc9 100644
--- a/DQMOffline/Trigger/python/MuonMonitor_cfi.py
+++ b/DQMOffline/Trigger/python/MuonMonitor_cfi.py
@@ -39,3 +39,8 @@ hltMuonmonitoring.denGenericTriggerEventPSet.andOrDcs      = cms.bool( False )
 hltMuonmonitoring.denGenericTriggerEventPSet.errorReplyDcs = cms.bool( True )
 hltMuonmonitoring.denGenericTriggerEventPSet.verbosityLevel = cms.uint32(1)
 
+from Configuration.Eras.Modifier_stage2L1Trigger_cff import stage2L1Trigger
+stage2L1Trigger.toModify(hltMuonmonitoring,
+                         numGenericTriggerEventPSet = dict(stage2 = cms.bool(True)),
+                         denGenericTriggerEventPSet = dict(stage2 = cms.bool(True)))
+
```

I don't want to own a PR about it and have no time to test it, but @tvami or @panoskatsoulis feel free to check it if works for you

**Comment by tvami** [comment]
Thanks Marco, I can submit the PR

**Comment by missirol** [comment]
unassign hlt

**Comment by tvami** [comment]
assign dqm

**Comment by tvami** [comment]
Yes, seems like you are right @missirol 
https://github.com/cms-sw/cmssw/pull/37033#issuecomment-1048084313

**Comment by missirol** [comment]
A quick workaround would be to add `stage2` to the `fillDescriptions` of those 2/3 HLT-DQM plugins (anyways, that's a DQM package, so not really up to me to comment).

**Comment by jfernan2** [comment]
+1
For the records

**Comment by jfernan2** [comment]
@iatakisi @ramankhurana  @physicist87 @bobaekim @pallabidas could you please advice in the issue above?
Thanks

**Comment by ramankhurana** [comment]
Thanks for notice, but I am not responsible for this anymore (from last 4 years).
